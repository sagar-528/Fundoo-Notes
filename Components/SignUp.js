import React from 'react'
import { View, Text, StyleSheet, TouchableWithoutFeedback, Keyboard, TouchableOpacity, TextInput } from 'react-native';
import * as Animatable from 'react-native-animatable';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';
import Feather from 'react-native-vector-icons/Feather';
import GlobalCss from '../Styles/GlobalCss'

const SignUp = ( { navigation } ) => {
  return(
    <TouchableWithoutFeedback onPress = {() => {
      Keyboard.dismiss();
    }}>
    <View style = {GlobalCss.container}>

      <View style = {GlobalCss.header}>
        <Text style = {GlobalCss.text_header}>Sign Up</Text>
      </View>

      <Animatable.View 
      animation = "fadeInUpBig"
      style = {GlobalCss.footer}
  >
      <Text style = {GlobalCss.text_footer}>Username</Text>
      
      <View style = {GlobalCss.action}>
          <FontAwesome5 
              name = "user"
              size = {20}
              color = "black"
          />
          <TextInput 
              placeholder = "Your Username"
              placeholderTextColor = "black"
              style = {GlobalCss.textInput}
          />
      </View>
      
      <Text style = {[GlobalCss.text_footer, {
          marginTop: 35
      }]}>Password</Text>
      
      <View style = {GlobalCss.action}>
          <Feather 
              name = "unlock"
              size = {20}
              color = "black"
          />
          <TextInput 
              placeholder = "Your Password"
              placeholderTextColor = "black"
              style = {GlobalCss.textInput}
           />
      </View>
      
      <Text style = {[GlobalCss.text_footer, {
        marginTop: 35
    }]}> Confirm Password </Text>
    
    <View style = {GlobalCss.action}>
        <Feather 
            name = "lock"
            size = {20}
            color = "black"
        />
        <TextInput 
            placeholder = "confirm Password"
            placeholderTextColor = "black"
            style = {GlobalCss.textInput}
         />
    </View>
    
          <View style = {{alignItems: 'center', marginTop: 15}}>
          <TouchableOpacity
              style = {[GlobalCss.signIn, {
                  borderColor: '#009387',
                  borderWidth: 1,
                  marginTop: 15
              }]}
              onPress = { () => {navigation.navigate('SignIn')} }
          >
              <Text style = {[GlobalCss.textSign, {
                  color: 'black'
              }]}>Sign Up</Text>
          </TouchableOpacity>
          
          <View style = {{flexDirection:'row',padding: 5}}>
            <Text style = {{color: 'black'}}> Already have an account? </Text>
            <TouchableOpacity onPress = { () => {navigation.navigate('SignIn')} }>
              <Text style = {{color: '#1976d2', fontSize: 16}}> Sign In</Text>
            </TouchableOpacity>
          </View>
      </View>
  </Animatable.View>

    </View>
    </TouchableWithoutFeedback>
  );
};

export default SignUp