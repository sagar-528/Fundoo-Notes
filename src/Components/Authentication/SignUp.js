import React, {Component} from 'react'
import { View, Text, TouchableWithoutFeedback, Keyboard, TouchableOpacity, TextInput, ScrollView } from 'react-native';
import * as Animatable from 'react-native-animatable';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';
import Feather from 'react-native-vector-icons/Feather';
import GlobalCss from '../../Styles/GlobalCss'
import UserServices from '../../../Service/UserServices';
import { Dialog, Portal, Button, Provider, Paragraph } from 'react-native-paper';

export default class SignUp extends Component{
  
  constructor(props) {
    super(props)
    
    this.state = {
        firstName : '',
        lastName : '',
        email : '',
        password : '',
        firstNameValidation : true,
        lastNameValidation : true,
        emailValidation : true,
        passwordValidation : true,
        secureTextPassword : true,
        firstNameEmpty : false,
        lastNameEmpty : false,
        emailEmpty : false,
        passwordEmpty : false,
        emailPresent : false,
        visible : false,
    }
  }

 textInputChangeFirstName = (val) =>{
  this.setState({
    firstName: val,
    firstNameEmpty : false
  })
  this.validateFirstName();
}

textInputChangeLastName = (val) =>{
    this.setState({
     lastName: val,
     lastNameEmpty : false
   })
   this.validateLastName();
 }

textInputChangeEmail = (val) =>{
    this.setState({
     email: val,
     emailEmpty : false,
     emailPresent : false
   })
   this.validateEmail();
 }

 textInputChangePassword = (val) =>{
    this.setState({
     password: val,
     passwordEmpty : false,
   })
   this.validatePassword();
 }

  validateFirstName = () => {
    const nameRegex = new RegExp("^[A-Z][a-z]{1,}$")
      if(nameRegex.test(this.state.firstName)) {
        this.setState({
          firstNameValidation : true
      })
    }
    else {
        this.setState({
          firstNameValidation : false
      })
    }
  }

  validateLastName = () => {
    const nameRegex = new RegExp("^[A-Z][a-z]{2,}$")
    if(nameRegex.test(this.state.lastName)) {
      this.setState({
          lastNameValidation : true
      })
    }
    else {
      this.setState({
          lastNameValidation : false
    })
   }
  }

validateEmail = () => {
  const emailRejex = new RegExp("^[0-9a-zA-Z]+([._+-][0-9A-Za-z]+)*@[0-9A-Za-z]+[.][a-zA-Z]{2,4}([.][a-zA-Z]{2,4})?$")
  if(emailRejex.test(this.state.email)) {
      this.setState({
          emailValidation : true
      })
  } 
  else {
      this.setState({
          emailValidation : false
      })
  }
}

validatePassword = () => {
  const passwordRegex = new RegExp("^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[*.!@#$%^&(){}:'<>,.>/~`_+=|].).{8,}$");
  if(passwordRegex.test(this.state.password)) {
      this.setState({
          passwordValidation: true
      })
  }
  else {
      this.setState({
          passwordValidation: false
      })
  }
}

 signUpHandler = async() =>{
  const {onPress} = this.props;
    if(this.state.firstName != '' &&  this.state.lastName != '' && this.state.email != '' && this.state.password != ''){
      UserServices.SignUp(this.state.email, this.state.password)
          .then( async userCredential => {
           await this.setState({
              visible : true
            })
            UserServices.writeUserDataInRealtimeDatabase(userCredential.user.uid, this.state.firstName, this.state.lastName, userCredential.user.email)
              this.props.navigation.push("SignIn")
          })
          .catch(error => {
              if(error == 'Email Already Exist') {
                  this.setState({
                      emailPresent : true
                  })
              }
          })
    }
    else{
      if(this.state.firstName == '') {
          await this.setState({
              firstNameEmpty : true
          })
      }
      if(this.state.lastName == ''){
          await this.setState({
              lastNameEmpty : true
          })
      }
      if(this.state.email == '') {
          await this.setState({
              emailEmpty : true
          })
      }
      if(this.state.password == '') {
          await this.setState({
              passwordEmpty : true
          })
      }
      if(this.state.confirmPassword == '') {
          await this.setState({
              confirmPasswordEmpty : true
          })
      }
  }   
  // onPress();
 }

 signInHandler = () => {
   const {onPress} = this.props;
    this.props.navigation.push("SignIn")
    // onPress();
 }

 hideDialog = async  () => {
  const {onDismiss} = this.props
   this.setState({
      visible : false
  })
  // onDismiss()
}

handleDialogButton = () => {
  const {onPress} = this.props
  this.props.navigation.navigate('SignIn')
  // onPress()
}

render(){
  return(
    <Provider>
    <ScrollView>
    <TouchableWithoutFeedback onPress = {() => {
      Keyboard.dismiss();
    }}>
    <View style = {GlobalCss.container}>

      <View style = {GlobalCss.header}>
        <FontAwesome5 
                name = "foursquare"
                color = "#fff"
                size = {50}
        />
        <Text style = {GlobalCss.text_header}>Fundoo Notes</Text>
      </View>

    <Animatable.View 
      animation = "fadeInUpBig"
      style = {GlobalCss.footer}
    >

    <Text style = {GlobalCss.text_footer}>First Name</Text>
    <View style = {GlobalCss.action}>
        <FontAwesome5 
            name = "user-circle"
            size = {20}
            color = "black"
        />
        <TextInput 
            placeholder = "First Name"
            placeholderTextColor = "black"
            style = {GlobalCss.textInput}
            onChangeText={this.textInputChangeFirstName}
        />
      </View>
      <View>
        <Text style = {GlobalCss.text_error_style}>
            {(this.state.firstNameEmpty) ? 'required..' : (this.state.firstNameValidation) ? null : 'Invalid First Name'}
        </Text>
      </View>
    <Text style = {[GlobalCss.text_footer, {
      marginTop: 10
      }]}>Last Name</Text>
    <View style = {GlobalCss.action}>
        <FontAwesome5 
            name = "user-circle"
            size = {20}
            color = "black"
        />
        <TextInput 
            placeholder = "Last Name"
            placeholderTextColor = "black"
            style = {GlobalCss.textInput}
            onChangeText={this.textInputChangeLastName}
        />
    </View>
      <View>
        <Text style = {GlobalCss.text_error_style}>
            {(this.state.lastNameEmpty) ? 'required..' : (this.state.lastNameValidation) ? null : 'Invalid Last Name..'}
        </Text>
      </View>
      <Text style = {[GlobalCss.text_footer, {
        marginTop: 10
        }]}>Email-Id</Text>
      <View style = {GlobalCss.action}>
          <FontAwesome5 
              name = "user"
              size = {20}
              color = "black"
          />
          <TextInput 
              placeholder = "Your Email-Id"
              placeholderTextColor = "black"
              style = {GlobalCss.textInput}
              onChangeText={this.textInputChangeEmail}
          />
      </View>
      <View>    
        <Text style = {GlobalCss.text_error_style}>
            {(this.state.emailEmpty) ? 'required..' : (this.state.emailValidation) ? (this.state.emailPresent) ? 'Email Already Exist' : null : 'Invalid Email..'}
        </Text>
      </View>  
      <Text style = {[GlobalCss.text_footer, {
          marginTop: 10
      }]}>Password</Text>
      
      <View style = {GlobalCss.action}>
          <Feather 
              name = "unlock"
              size = {20}
              color = "black"
          />
          <TextInput 
              placeholder = "Your Password"
              placeholderTextColor = "black"
              style = {GlobalCss.textInput}
              onChangeText={this.textInputChangePassword}
              secureTextEntry = {this.state.secureTextPassword}
           />
      </View>
      <View>    
        <Text style = {GlobalCss.text_error_style}>
            {(this.state.passwordEmpty) ? 'required..' : (this.state.passwordValidation) ? null : 'Invalid Password..'}
        </Text>
      </View>
          <View style = {{alignItems: 'center', marginTop: 20}}>
          <TouchableOpacity
              style = {[GlobalCss.signIn, {
                  borderColor: '#009387',
                  borderWidth: 1,
                  marginTop: 15
              }]}
              onPress = { this.signUpHandler }
          >
              <Text style = {[GlobalCss.textSign, {
                  color: 'black'
              }]}>Sign Up</Text>
          </TouchableOpacity>
          
          <View style = {{flexDirection:'row',padding: 5}}>
            <Text style = {{color: 'black', fontSize: 17}}>Already have an account? </Text>
            <TouchableOpacity onPress = { this.signInHandler }>
              <Text style = {{color: '#1976d2', fontSize: 17}}>Sign In</Text>
            </TouchableOpacity>
          </View>
       </View>
       <Portal>
          <Dialog visible={this.state.visible} onDismiss={this.hideDialog}>
            <Dialog.Content>
              <Paragraph style = {{fontSize : 20, paddingTop : 20}}>SucessFull Sign Up.</Paragraph>
            </Dialog.Content>
            <Dialog.Actions>
              <Button onPress={this.handleDialogButton}>Ok</Button>
            </Dialog.Actions>
          </Dialog>
        </Portal>
      </Animatable.View>
    </View>
    </TouchableWithoutFeedback>
    </ScrollView>
    </Provider>
  )}
}
