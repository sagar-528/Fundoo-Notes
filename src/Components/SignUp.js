import React from 'react'
import { View, Text, TouchableWithoutFeedback, Keyboard, TouchableOpacity, TextInput } from 'react-native';
import * as Animatable from 'react-native-animatable';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';
import Feather from 'react-native-vector-icons/Feather';
import GlobalCss from '../Styles/GlobalCss'

const SignUp = ( { navigation } ) => {

const signUpHandler = () => {
  navigation.navigate('SignIn')
}

const signInHandler = () => {
  navigation.navigate('SignIn')
}
  return(
    <TouchableWithoutFeedback onPress = {() => {
      Keyboard.dismiss();
    }}>
    <View style = {GlobalCss.container}>

      <View style = {GlobalCss.header}>
        <FontAwesome5 
                name = "google"
                color = "#fff"
                size = {40}
        />
        <Text style = {GlobalCss.text_header}>Fundoo-Notes</Text>
      </View>

    <Animatable.View 
      animation = "fadeInUpBig"
      style = {GlobalCss.footer}
    >

    <Text style = {GlobalCss.text_footer}>First Name</Text>
    <View style = {GlobalCss.action}>
        <FontAwesome5 
            name = "user"
            size = {20}
            color = "black"
        />
        <TextInput 
            placeholder = "Your First Name"
            placeholderTextColor = "black"
            style = {GlobalCss.textInput}
        />
    </View>

    <Text style = {[GlobalCss.text_footer, {
      marginTop: 10
      }]}>Last Name</Text>
    <View style = {GlobalCss.action}>
        <FontAwesome5 
            name = "user"
            size = {20}
            color = "black"
        />
        <TextInput 
            placeholder = "Your Last Name"
            placeholderTextColor = "black"
            style = {GlobalCss.textInput}
        />
    </View>

      <Text style = {[GlobalCss.text_footer, {
        marginTop: 10
        }]}>Username</Text>
      <View style = {GlobalCss.action}>
          <FontAwesome5 
              name = "user"
              size = {20}
              color = "black"
          />
          <TextInput 
              placeholder = "Your Username"
              placeholderTextColor = "black"
              style = {GlobalCss.textInput}
          />
      </View>
      
      <Text style = {[GlobalCss.text_footer, {
          marginTop: 10
      }]}>Password</Text>
      
      <View style = {GlobalCss.action}>
          <Feather 
              name = "unlock"
              size = {20}
              color = "black"
          />
          <TextInput 
              placeholder = "Your Password"
              placeholderTextColor = "black"
              style = {GlobalCss.textInput}
           />
      </View>
    
          <View style = {{alignItems: 'center', marginTop: 15}}>
          <TouchableOpacity
              style = {[GlobalCss.signIn, {
                  borderColor: '#009387',
                  borderWidth: 1,
                  marginTop: 15
              }]}
              onPress = { signUpHandler }
          >
              <Text style = {[GlobalCss.textSign, {
                  color: 'black'
              }]}>Sign Up</Text>
          </TouchableOpacity>
          
          <View style = {{flexDirection:'row',padding: 5}}>
            <Text style = {{color: 'black', fontSize: 17}}> Already have an account? </Text>
            <TouchableOpacity onPress = { signInHandler }>
              <Text style = {{color: '#1976d2', fontSize: 17}}> Sign In</Text>
            </TouchableOpacity>
          </View>
      </View>
  </Animatable.View>

    </View>
    </TouchableWithoutFeedback>
  );
};

export default SignUp