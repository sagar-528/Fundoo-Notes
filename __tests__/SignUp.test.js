import React from 'react';
import {shallow} from 'enzyme';
import SignUp from '../src/Components/SignUp'

describe('test SignUp', () => {
    
    it('should match to snapshot', () => {
        const component = shallow(<SignUp />)
        expect(component).toMatchSnapshot();
    })

    it('test when first name provided in textinput should update first name state', async () => {
        const component = shallow(<SignUp />)
        expect(component.instance().state.firstName).toBe('')
        component.instance().textInputChangeFirstName('Sagar')
        expect(component.instance().state.firstName).toBe('Sagar')
    })

    it('test when last name provided in textinput should update last name state', async () => {
        const component = shallow(<SignUp />)
        expect(component.instance().state.lastName).toBe('')
        component.instance().textInputChangeLastName('Gupta')
        expect(component.instance().state.lastName).toBe('Gupta')
    })

    it('test when email provided in textinput should update email state', async () => {
        const component = shallow(<SignUp />)
        expect(component.instance().state.email).toBe('')
        component.instance().textInputChangeEmail('gupta.sagar528@gmail.com')
        expect(component.instance().state.email).toBe('gupta.sagar528@gmail.com')
    })

    it('test when password provided in textinput should update password state', async () => {
        const component = shallow(<SignUp />)
        expect(component.instance().state.password).toBe('')
        component.instance().textInputChangePassword('Qwerty@111')
        expect(component.instance().state.password).toBe('Qwerty@111')
    })

    it('test when provided firstName is valid firstName validation state will be true', async () => {
        const component = shallow(<SignUp />)
        component.instance().setState({ firstName : 'Sagar'})
        component.instance().validateFirstName()
        expect(component.instance().state.firstNameValidation).toBe(true)
    })

    it('test when provided firstName is invalid firstName validation state will be false', async () => {
        const component = shallow(<SignUp />)
        component.instance().setState({ firstName : 'sagar'})
        component.instance().validateFirstName()
        expect(component.instance().state.firstNameValidation).toBe(false)
    })

    it('test when provided lastName is valid lastName validation state will be true', async () => {
        const component = shallow(<SignUp />)
        component.instance().setState({ lastName : 'Gupta'})
        component.instance().validateLastName()
        expect(component.instance().state.lastNameValidation).toBe(true)
    })

    it('test when provided lastName is invalid lastName validation state will be false', () => {
        const component = shallow(<SignUp />)
        component.instance().setState({ lastName : 'gupta'})
        component.instance().validateLastName()
        expect(component.instance().state.lastNameValidation).toBe(false)
    })

    it('test when provided email is valid email validation state will be true', async () => {
        const component = shallow(<SignUp />)
        component.instance().setState({ email : 'gupta.sagar528@gmail.com'})
        component.instance().validateEmail();
        expect(component.instance().state.emailValidation).toBe(true)
    })

    it('test when provided email is invalid email validation state will be false', async () => {
        const component = shallow(<SignUp />)
        component.instance().setState({ email : 'gupta.sagar528#gmail.com'})
        component.instance().validateEmail();
        expect(component.instance().state.emailValidation).toBe(false)
    })

    it('test when provided password is valid password validation state will be true', async () => {
        const component = shallow(<SignUp />)
        component.instance().setState({ password : 'Qwerty@111'})
        component.instance().validatePassword();
        expect(component.instance().state.passwordValidation).toBe(true)
    })

    it('test when provided password is invalid password validation state will be false', async () => {
        const component = shallow(<SignUp />)
        component.instance().setState({ password : 'qwerty1234'})
        component.instance().validatePassword();
        expect(component.instance().state.passwordValidation).toBe(false)
    })

    it('test onPress event of sign in button it will navigate to sign in screen', async () => {
        const navigation = { push : jest.fn() }
        const onPressEvent = jest.fn();
        const component = shallow(<SignUp onPress = {onPressEvent} navigation = {navigation}/>)
        const instance = component.instance();
        instance.signInHandler();
        expect(onPressEvent).toHaveBeenCalled();
        expect(navigation.push).toBeCalledWith("SignIn");
    })

    it('test onPress event of sign up button when all textinput empty it will update the stateEmpty state to true ', async () => {
        const onPressEvent = jest.fn();
        const component = shallow(<SignUp onPress = {onPressEvent}/>)
        const instance = component.instance();
        
        await instance.signUpHandler();
        expect(onPressEvent).toHaveBeenCalled();
        expect(instance.state.firstNameEmpty).toBe(true)
        expect(instance.state.lastNameEmpty).toBe(true)
        expect(instance.state.emailEmpty).toBe(true)
        expect(instance.state.passwordEmpty).toBe(true)
    })

})
