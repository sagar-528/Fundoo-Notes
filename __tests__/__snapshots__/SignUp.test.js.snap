// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Register should match to snapshot 1`] = `ShallowWrapper {}`;

exports[`test SignUp should match to snapshot 1`] = `ShallowWrapper {}`;
